### DISCUSSION TITLE
Fill the field automatically

### DISCUSSION THREAD

**Discussion by sandhitya069** [serenity-is/Serenity#5488] [Help]
Hi Everyone,
i want to fill the state and country automatically when i select the city, but it's doesn't work
state and country still blank.
![image](https://user-images.githubusercontent.com/55977575/108026971-5af55480-7064-11eb-813b-7c3d8d107aba.png)

here my PersonRow.cs
`[DisplayName("City"), NotNull,
        ForeignKey("[dbo].[City]", "Oid"),
        LeftJoin("jCity"), TextualField("CityName")]
        [LookupEditor(typeof(CityRow),
        InplaceAdd = true),
        LookupInclude]
        public int? City
        {
            get { return Fields.City[this]; }
            set { Fields.City[this] = value; }
        }

        [DisplayName("City"),
            Expression("jCity.[CityName]")]
        public String CityName
        {
            get { return Fields.CityName[this]; }
            set { Fields.CityName[this] = value; }
        }

        [DisplayName("State")]
        [Expression("jCity.[State]")]
        [ForeignKey("State", "Oid")]
        [LeftJoin("jState")] //, TextualField("jState.[StateName]")]
        [ReadOnly(true)]
        public int? State
        {
            get { return Fields.State[this]; }
            set { Fields.State[this] = value; }
        }

        [DisplayName("State")]
        [Expression("jState.[StateName]")]
        [ReadOnly(true)]
        public String StateName
        {
            get { return Fields.StateName[this]; }
            set { Fields.StateName[this] = value; }
        }

        [DisplayName("Country")]
        [Expression("jState.[Country]")]
        [ForeignKey("Countries", "Oid")]
        [LeftJoin("jCountry")]//, TextualField("jCountry.[CountryName]")]
        [ReadOnly(true)]
        public int? Country
        {
            get { return Fields.Country[this]; }
            set { Fields.Country[this] = value; }
        }

        [DisplayName("Country")]
        [Expression("jCountry.[CountryName]")]
        [ReadOnly(true)]
        public String CountryName
        {
            get { return Fields.CountryName[this]; }
            set { Fields.CountryName[this] = value; }
        }`

here my PersonForm.cs
`[Category("Address")]
        public string Address { get; set; }
        [HalfWidth]
        public string City { get; set; }
        [HalfWidth]
        public string State { get; set; }
        [HalfWidth]
        public string Country { get; set; }
        [HalfWidth]
        public int PostalCode { get; set; }
        [HalfWidth]
        public int Phone { get; set; }
        [HalfWidth]
        public int Fax { get; set; }
`

here my PersonColumn.cs
` [QuickFilter]
        [Width(250)]
        public string CityName { get; set; }

        [QuickFilter]
        [Width(250)]
        public string StateName { get; set; }

        [QuickFilter]
        [Width(250)]
        public string CountryName { get; set; }`
someone please tell me something to do

**Comment by luongnv6** [comment]
use City Change2select event in dialog.ts

**Reply by sandhitya069** [reply]
can you give me some example please or link tutorial maybe @luongnv6

**Reply by luongnv6** [reply]
this is code demo:
https://github.com/serenity-is/Serene/blob/8f0f778d1331d3988610854e0a85e14dbff20757/Serene/Serene.Core/Modules/Northwind/OrderDetail/OrderDetailDialog.ts#L17

**Reply by sandhitya069** [reply]
okaayy, thanks sir @luongnv6
